cranelift/codegen/src/isle_prelude.rs

```diff
@@ -49,6 +59,28 @@ macro_rules! isle_common_prelude_methods {
             Some(Imm64::new(result))
         }
 
+        #[inline]
+        fn imm64_srem(&mut self, ty: Type, x: Imm64, y: Imm64) -> Option<Imm64> {
+            // Sign extend `x` and `y`.
+            let shift = u32::checked_sub(64, ty.bits()).unwrap_or(0);
```

Comment:
I'm curious why the `checked_sub` is necessary here, rather than an assert that `ty.bits() <= 64` (which should be the case for any `Imm64`)?